utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,218 @@
+                }
+            }
+        }
+    }
+    return telecom;
+}
+
+public isolated function mapCcdaNametoFhirName(xml<xml:Element> nameElement, boolean isNamespaceAvailable = true) returns r4:HumanName|error? {
+    if nameElement is xml:Element && nameElement.length() > 0 {
+        r4:HumanName name = {};
+        xml<xml:Element> familyElement;
+        xml<xml:Element> givenElement;
+        if isNamespaceAvailable {
+            familyElement = nameElement.children().<v3:family>;
+            givenElement = nameElement.children().<v3:given>;
+        } else {
+            familyElement = nameElement.children().<family>;
+            givenElement = nameElement.children().<given>;
+        }
+        if familyElement is xml:Element && familyElement.length() > 0 {
+            name.family = familyElement.get(0).data();
+        }
+
+        if givenElement is xml:Element && givenElement.length() > 0 {
+            name.given = [givenElement.get(0).data()];
+        }
+        return name;
+    }
+    return;
+}
+
+public isolated function mapCcdaCodingtoFhirCodeableConcept(xml<xml:Element> codingElement) returns r4:CodeableConcept|error {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        r4:CodeableConcept codeableConcept = {};
+        r4:Coding mapCCDACodingtoFHIRCodeResult = check mapCcdaCodingtoFhirCode(codingElement);
+        codeableConcept.coding = [mapCCDACodingtoFHIRCodeResult];
+        return codeableConcept;
+    }
+    return {};
+}
+
+public isolated function mapCcdaCodingtoFhirCode(xml<xml:Element> codingElement) returns r4:Coding|error {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        r4:Coding coding = {};
+        string|error codeVal = codingElement.code;
+        if codeVal is string {
+            coding.code = codeVal;
+        }
+        string|error systemVal = codingElement.codeSystem;
+        if systemVal is string {
+            coding.system = systemVal;
+        }
+        return coding;
+    }
+    return {};
+}
+
+public isolated function mapCcdaGenderCodetoFhirGender(xml<xml:Element> codingElement) returns r4:PatientGender|error? {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        string|error codeVal = codingElement.code;
+        if codeVal is string {
+            match codeVal {
+                "M" => {
+                    return r4:CODE_GENDER_MALE;
+                }
+                "F" => {
+                    return r4:CODE_GENDER_FEMALE;
+                }
+                "UN" => {
+                    return r4:CODE_GENDER_OTHER;
+                }
+                _ => {
+                    return r4:CODE_GENDER_UNKNOWN;
+                }
+            }
+        }
+    }
+    return;
+}
+
+public isolated function mapCcdaValueToFhirCategory(xml:Text codingElement) returns r4:AllergyIntoleranceCategory|error? {
+    string|error codeVal = codingElement.data();
```

Comment:
Don't we need to handle the error case?. We can use `check` expression to pass the error to the caller. Check similar cases

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/bd15548be31a7e76b4962973508cdedd6f8fd58a